### PR TITLE
IGNITE-19919: org.apache.ignite.sql.ResultSet#close should close implicit transaction if any.

### PR DESCRIPTION
**Pull request by lowka**
- Commits implicit transaction on AsyncCursor::close.
- Adds unit tests for AsyncCursorImpl.

https://issues.apache.org/jira/browse/IGNITE-19919

----

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Review by AMashenkov — APPROVED** [review]

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlSynchronousApiTest.java
Comment (xtern) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlSynchronousApiTest.java:1]: Please don't forget to check this TODO https://github.com/apache/ignite-3/blob/6d2dda7ca2ba7d0610d418a5ad9c321c06449ba1/modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlSynchronousApiTest.java#L308

**Comment by lowka** [comment]
@xtern Thanks. Fixed.

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/AsyncSqlCursorImplTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * Tests for {@link AsyncSqlCursorImpl}.
+ */
+@ExtendWith(MockitoExtension.class)
```
Comment (xtern) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/AsyncSqlCursorImplTest.java:48]: Do we really need mockito extension for this class?

Comment (lowka) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/AsyncSqlCursorImplTest.java]: Removed. Thanks.

**Pull request merged**